packages/react-widgets/src/widgets/LegendWidget.js

```diff
@@ -1,19 +1,29 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { updateLayer } from '@carto/react-redux/';
 import { LegendWidgetUI } from '@carto/react-ui';
 import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
 
-export default function LegendWidget({ className }) {
+/**
+ * Renders a <LegendWidget /> component
+ * @param  {object} props
+ * @param  {string} props.className - CSS class name
+ * @param  {string} props.initialCollapsed - Define initial collapsed value. false by default.
+ */
+function LegendWidget({ className, initialCollapsed }) {
   const dispatch = useDispatch();
   const layers = useSelector((state) =>
-    Object.values(state.carto.layers).filter((layer) => !!layer.legend)
+    [...Object.values(state.carto.layers), ...Object.values(state.carto.layers)].filter(
```

Comment:
Could you explain what are you doing here? Maybe you can create a selector to be more specific

Reply:
Ups, sorry. That was for testing. I'm going to remove it now

Reply:
Updated.